modules/packages/LinearAlgebra.chpl

```diff
@@ -221,6 +222,267 @@ class LinearAlgebraError : Error {
     }
 }
 
+/*
+This class promotes lazy computation of required
+Helper matrices to evaluate matrix exponential.
+
+High precision of expm requires smaller computations,
+hence we do a lazy computation depending on the norm value
+of the matrix.
+
+*/
+class ExpmPadeHelper {
+  type t;
+  var A : t;
+  var A2 : t;
+  var A4 : t;
+  var A6 : t;
+  var A8 : t;
+  var A10 : t;
+  var Exactd4 : real;
+  var Exactd6 : real;
+  var Exactd8 : real;
+  var Exactd10 : real;
+  var Approxd4 : real;
+  var Approxd6 : real;
+  var Approxd8 : real;
+  var Approxd10 : real;
+  var ident : t;
+  var use_exact_onenorm : bool;
+  // Boolean arrays to know if A, Exactd, Approxd
+  // have been computed.
+  var isAComputed : [1..5] bool;
+  var isExactdComputed : [1..4] bool;
+  var isApproxdComputed : [1..4] bool;
+
+  /*
+  Params:
+  A: Expects an N*N square matrix
+  use_exact_onenorm: boolean value specifying if the onenorm has to be exact.
+
+  TODO: Need to write a method to find approximate norm.
```

Comment:
Is this TODO still a todo item?  Or was it resolved and not updated?

Reply:
Yes this is still a Todo item, we decided to take that up (if necessary) in second PR. I'm considering to leave it as is, if that is fine? :)

Reply:
Yup, that's fine!  Just make sure there's an issue tracking it, in case we get distracted and forget about it